### PR TITLE
WIP: Add doctests for strings

### LINKED ISSUE #23392: Add doctest examples to some of the stdlib string functions
Scrolling through [the documentation](https://docs.julialang.org/en/stable/stdlib/strings/) we can see that many of the string functions (especially ones involving regexes) don't have any doctest examples. Adding some would be a great way to get started if you're looking to contribute!

If the docstring mentions several possible behaviours of a function (for instance, if there's a kwarg) it's nice to provide examples for each of the described behaviours.

### PR DESCRIPTION
**Pull request by iglpdc**
Fixes #23392.

### CODE REVIEW

File: base/strings/basic.jl [hunk 1]
```diff
@@ -100,7 +100,40 @@ function length(s::AbstractString)
 end
 
 ## string comparison functions ##
+"""
+    cmp(a::AbstractString, b::AbstractString)
+
+Compares two strings for equality. Returns `0` if both strings have the same
```
Comment (fredrikekre) [inline base/strings/basic.jl]: Just a general comment: docstrings should generally use imperative form, see point 2 here: https://docs.julialang.org/en/latest/manual/documentation/#Documentation-1 so `Compares` -> `Compare` etc

**Comment by iglpdc** [comment]
@fredrikekre do you want me to squash all these commits? (I guess I should have done before pushing: I'll do it better next time.)

**Comment by ararslan** [comment]
> do you want me to squash all these commits?

No need, we can squash the commits on merge.

File: base/strings/io.jl [hunk 2]
```diff
@@ -143,6 +156,16 @@ end
     repr(x)
 
 Create a string from any value using the [`show`](@ref) function.
+
+#Examples
```
Comment (fredrikekre) [inline base/strings/io.jl]: Missing space.

**Review by fredrikekre — COMMENTED** [review]
The docstrings which are new should be added to the stdlib. Seems like `doc/src/stdlib/strings.md` is the right place.

**Pull request merged**